mtgjson5/providers/tcgplayer.py

```
@@ -250,6 +281,43 @@ def get_tcgplayer_sku_data(group_id_and_name: Tuple[str, str]) -> List[Dict[str,
     return magic_set_product_data
 
 
+def get_tcgplayer_sealed_data(group_id: Optional[Any]) -> List[Dict[str, Any]]:
+    """
+    Finds all sealed product for a given group
+    :param group_id: group id for the set to get data for
+    :return: sealed product data with extended fields
+    """
+    magic_set_sealed_data = []
+    api_offset = 0
+
+    while True:
+        api_response = TCGPlayerProvider().download(
+            "https://api.tcgplayer.com/catalog/products",
+            {
+                "offset": str(api_offset),
+                "limit": 100,
+                "categoryId": 1,
+                "getExtendedFields": True,
+                "groupId": str(group_id),
+                "productTypes": "Booster Box,Booster Pack,Sealed Products",
+            },
+        )
+
+        if not api_response:
+            # No more entries
+            break
+
+        response = json.loads(api_response)
+        if not response["results"]:
+            # Something went wrong
```

maybe return the error array somewhere?

---

Should log an error